### PR TITLE
Add a test for explicit payload member and empty body responses

### PR DESCRIPTION
**Pull request by sugmanue**
#### Background

Add a new protocol tests for the JSON REST protocol. When we have a response object with a member that's bound to the HTTP payload, such as

```smithy
@output
structure HttpPayloadWithStructureInputOutput {
    @httpPayload
    nested: NestedPayload,
}
```

And the response body is empty, then the member MUST be set to `null`.

Regression for this case was reported to the Java SDK V2 and fixed on [#6111](https://github.com/aws/aws-sdk-java-v2/pull/6111).

#### Testing
* How did you test these changes?

#### Links
* Links to additional context, if necessary
* Issue #, if applicable (see [here](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) for a list of keywords to use for linking issues)

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: smithy-aws-protocol-tests/model/restJson1/http-payload.smithy [hunk 1]
```diff
@@ -281,6 +281,20 @@ apply HttpPayloadWithStructure @httpResponseTests([
...
 ])
 
+apply HttpPayloadWithStructure @httpResponseTests([
+    {
+        id: "RestJsonHttpPayloadWithStructureAndEmptyResponseBody",
+        documentation: "Serializes a structure in the payload",
+        protocol: restJson1,
+        code: 200,
+        body: "",
+        bodyMediaType: "application/json",
```
Comment (mtdowling) [inline smithy-aws-protocol-tests/model/restJson1/http-payload.smithy:291]: Not sure this would be set if there's no payload, or at least, you don't want to enforce it for servers do you?

Comment (sugmanue) [reply smithy-aws-protocol-tests/model/restJson1/http-payload.smithy:291]: Yeah, probably not, let me get rid of it.

**Review by mtdowling — APPROVED** [review]
